libraries/platforms/aws-fpga/bsg_manycore_platform.cpp

```diff
@@ -361,6 +362,9 @@ static int hb_mc_platform_get_credits(hb_mc_manycore_t *mc,
 
 
 
+// This track active manycore machine IDs
+static std::set<hb_mc_manycore_id_t> active_ids;
```

Comment:
I'm not sure I would implement this at the platform level because now all platforms will need to implement code that looks quite a bit like this. Implementing this at the manycore level would reduce code duplication.

Reply:
Max and I had a long friday afternoon debate about this. 

The answer is yes. This is the right way to do things, but it would require some re-engineering. Since this is small potatoes compared to everything else I want/need to do I'm going to merge this.

However, we can file an issue if anyone feels strongly

Reply:
OOC what's a machine id?

Reply:
A machine ID was originally intended for identifying different AWS FPGAs on an instance with multiple FPGAs. 

So, different, unique instances of a manycore managed by the runtime. 

In Verilator, it could be multiple independent instances of simulation (though I haven't tested it and explicitly disallow it)

Does that answer your question?